pythermalcomfort/charts/pmv_chart.py

```diff
@@ -0,0 +1,327 @@
+from math import ceil, floor
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+from scipy.optimize import brentq
+
+from pythermalcomfort.models import pmv
+from pythermalcomfort.utilities import v_relative, clo_dynamic, psy_ta_rh
+from pythermalcomfort.charts.theme import (
+    tight_margins,
+    index_mapping_dictionary,
+)
+
+
+def pmv_chart(df, pmv_constants, show_summary=False, si_ip="si"):
+    """Generates a psychrometric chart with the Predicted Mean Vote (PMV) comfort zone, ASHRAE 55.
+
+    Parameters
+    ----------
+    df : DataFrame
+        A DataFrame containing the input data. Must have the following columns:
+        'ta' (Dry bulb temperature [°C]), 'rh' (Relative humidity [%]).
+
+    pmv_constants : dict
+        A dictionary containing the constants required for the PMV calculation. It must have the following keys
+        'tr' (Mean radiant temperature [°C]), 'v' (Air speed [m/s]), 'met' (Metabolic rate [met]), 'clo' (Clothing
+        insulation [clo]).
+
+    show_summary : bool, optional
+        If True, a summary of the data is shown. Default is False.
+
+    si_ip : str, optional
+        The unit system to use. It can be 'si' (default) or 'ip'.
+
+    Returns
+    -------
+    fig : graph_objects.Figure
+        Plotly figure object.
+
+    """
+
+    var = "pmv"
+    var_metadata = index_mapping_dictionary[var]
+    var_name, var_unit, var_range = (
+        var_metadata["name"],
+        var_metadata[si_ip]["unit"],
+        var_metadata[si_ip]["range"],
+    )
+    cat_color, cat_order = (
+        var_metadata["colors_categories"],
+        var_metadata["order_categories"],
+    )
+
+    t_dry = np.linspace(10, 36, 500)
+    rh_levels = np.linspace(0, 100, 10)
+    tr, v, met, clo = (
+        pmv_constants["tr"],
+        pmv_constants["v"],
+        pmv_constants["met"],
+        pmv_constants["clo"],
+    )
+
+    vr = v_relative(v=v, met=met)
+    clo_d = clo_dynamic(clo=clo, met=met)
+
+    observations = {
+        "t_dry": df["ta"].to_list(),
+        "rh": df["rh"].to_list(),
+    }
+
+    # helper functions start
+
+    def pmv_eq(T, RH):
```

Comment:
In pythons to the variable name should not be capitalized

Reply:
Do we really need this function to be defined here? can't we just have the PMV function in line 84?